ibis/expr/api.py

```diff
@@ -4193,27 +4193,60 @@ def mutate(table, exprs=None, **mutations):
         for name, expr in sorted(mutations.items(), key=operator.itemgetter(0))
     )
 
+    selection_exprs_by_name = collections.OrderedDict()
     for expr in exprs:
-        if expr.get_name() and isinstance(expr, ir.DestructColumn):
-            raise com.ExpressionError(
-                f"Cannot name a destruct column: {expr.get_name()}"
-            )
+        if isinstance(expr, ir.DestructColumn):
+            if expr.get_name():
+                raise com.ExpressionError(
+                    f"Cannot name a destruct column: {expr.get_name()}"
+                )
+            struct_type = expr.type()
+            for name in struct_type.names:
+                selection_exprs_by_name[name] = expr
```

Comment:
do we have a test that exercises both of these branches in the same? e.g. a mutate with a name that hits a udf name? (weird but ....)

Reply:
Just added a test test_elementwise_udf_overwrite_destruct_and_assign that should hit both of these branches.